gen/namespace.go

```diff
@@ -112,7 +112,7 @@ func (n *namespace) Rotate(base string) (string, error) {
 }
 
 func (n *namespace) Reserve(name string) error {
-	if n.isTaken(name) {
+	if n.isTaken(name) && (name != "init") {
```

Comment:
I think the init special case should stand out and be explained.

```suggestion
	// A single Go file can have multiple init() functions so we don't need to
	// reserve it.
	if name == "init" {
		return nil
	}

	if n.isTaken(name) {
```

Reply:
Adjusted.